ffi/diplomat/src/zoned_formatter.rs

```diff
@@ -0,0 +1,130 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+#[diplomat::bridge]
+pub mod ffi {
+    use alloc::boxed::Box;
+    use diplomat_runtime::DiplomatResult;
+    use icu_calendar::{Date, DateTime, Gregorian};
+    use icu_datetime::{options::length, TypedZonedDateTimeFormatter, ZonedDateTimeFormatter};
+
+    use crate::{
+        date::ffi::{ICU4XDate, ICU4XIsoDate},
+        datetime::ffi::ICU4XDateTime,
+        datetime::ffi::ICU4XIsoDateTime,
+        datetime_formatter::ffi::ICU4XDateLength,
+        datetime_formatter::ffi::ICU4XTimeLength,
+        errors::ffi::ICU4XError,
+        locale::ffi::ICU4XLocale,
+        provider::ffi::ICU4XDataProvider,
+        time::ffi::ICU4XTime,
+        timezone::ffi::ICU4XCustomTimeZone,
+        timezone_formatter::ffi::ICU4XIsoTimeZoneOptions,
+        timezone_formatter::ffi::ICU4XIsoTimeZoneFormat,
+        timezone_formatter::ffi::ICU4XIsoTimeZoneMinuteDisplay,
+        timezone_formatter::ffi::ICU4XIsoTimeZoneSecondDisplay,
+    };
+
+    #[diplomat::opaque]
+    /// An object capable of formatting a date time with time zone to a string.
+    #[diplomat::rust_link(icu::datetime::ZonedDateTimeFormatter, Struct)]
+    pub struct ICU4XZonedDateTimeFormatter(pub ZonedDateTimeFormatter);
+
+    impl ICU4XZonedDateTimeFormatter {
+        /// Creates a new [`ICU4XZonedDateTimeFormatter`] from locale data.
```

Comment:
nit: I would include docs about the method split since that's not on the original docs. Perhaps just mention the enum.